### PR TITLE
Add Org.qgis.qgis-ltr

### PR DESCRIPTION
**Pull request by hristo-mavrodiev**
The Org.qgis.qgis contains the latest version of QGIS. This version is not a "production" environment for GIS projects. Purpose of this repo is to add the latest stable LTR version of QGIS. For example in Arch linux there are qgis and qgis-ltr packages - https://www.archlinux.org/packages/community/x86_64/qgis/  and https://aur.archlinux.org/packages/qgis-ltr/?O=20&PP=10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## The LTR version of QGIS copied from https://github.com/flathub/org.qgis.qgis
```
Comment (barthalion) [inline README.md:1]: There's no point in having it here. No user is going to ever read it.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    "only-arches": ["aarch64", "x86_64"],
+    "skip-appstream-check": true
```
Comment (barthalion) [inline flathub.json:3]: Appdata should be fixed instead of disabling the check, and if it doesn't already, it should mention it's an LTR version.

**Review by barthalion — CHANGES_REQUESTED** [review]

**Comment by barthalion** [comment]
How common is it to prefer LTR version over the latest one? Also for the record, Arch ships only latest QGIS. AUR is free-for-all thing.

cc @aleixpol

**Comment by barthalion** [comment]
bot, build org.qgis.qgis-ltr

**Comment by hristo-mavrodiev** [comment]
bot, build org.qgis.qgis-ltr

**Comment by hristo-mavrodiev** [comment]
bot, build org.qgis.qgis-ltr

**Comment by aleixpol** [comment]
> How common is it to prefer LTR version over the latest one? Also for the record, Arch ships only latest QGIS. AUR is free-for-all thing.
> 
> cc @aleixpol

I was planning to look into it as well. My plan was to have it in a branch for org.qgis.qgis rather than a separate repo. If someone else to maintain it, that works too.

I asked QGIS maintainers and they recommended to prefer this version to the latest. I'm guessing it's because of the extended use of the python scripting this system does.

I can commit to keep this LTR (or a branch, I actually asked in flatpak@ mailing list about this already) up to date if you want together with the stable version if someone wants.

It feels a bit odd that as is here, they will be listed as separate applications.

File: org.qgis.qgis-ltr.json [hunk 3]
```diff
@@ -0,0 +1,614 @@
+{
+    "id" : "org.qgis.qgis-ltr",
+    "branch" : "stable",
+    "base": "io.qt.qtwebkit.BaseApp",
+    "base-version": "5.12",
+    "runtime" : "org.kde.Platform",
+    "runtime-version" : "5.12",
```
Comment (LeandroStanger) [inline org.qgis.qgis-ltr.json:7]: Update runtime to 5.15
```suggestion
    "base-version": "5.15",
    "runtime" : "org.kde.Platform",
    "runtime-version" : "5.15",
```

**Comment by barthalion** [comment]
> I was planning to look into it as well. My plan was to have it in a branch for org.qgis.qgis rather than a separate repo. If someone else to maintain it, that works too.

It depends what you want to achieve. If LTR is pushed as a separate branch of existing QGIS, only one will be visible in the system application list. The other can be still launched from CLI with `flatpak run --branch=foobar` or made default via `flatpak make-current`.

This also means that LTR and the latest releases will share configuration directories. Unless they're already versioned by the app, it may (or not! I don't use it) cause some issues if they are not compatible.

In such situations separate app id is IMHO preferred but I leave the decision to you.

**Comment by hristo-mavrodiev** [comment]
Sometimes the QGIS config is messed up (even when not switching versions) - QGIS fails to start, some bugs appear also.By default QGIS configs are stored in "/home/USER/.local/share/QGIS/QGIS3/"
The configuration files when using flatpak are store in   "$HOME/.var/app/org.qgis.qgis/data/QGIS/QGIS3"
Just deleting "org.qgis.qgis" folder will solve the issue and QGIS creates new configuration files. It's like a fresh new install.
The proposed build for LTR version fails, because it's now searching for qgis-ltr executable(desktop icon) instead of qgis. I have no idea how to solve this.
As a GIS user what I think that other people will love to see is to have the ability to choose a specific version for QGIS, like they have this choice on WinOS or MacOS. Also to have the same version of the important GIS dependencies (Qt, GDAL/OGR, GEOS, SQLite, PostgreSQL Client Version, QWT Version, PROJ, SpatiaLite Version, QScintilla2 Version, GRASS, SAGA,OracleDB driver, MSSQL DB driver) between the existing windows built and flatpak. So no matter you are using Linux or Windows you will have same result for this QGIS version.  It will be great if we could provide same experience as choosing an org.kde.sdk for example:
kde sdk
I will try to help you, but I don't have the knowledge to fully maintain the LTR version. Probably to stay with org.qgis.qgis will be better...

**Comment by eszlari** [comment]
I would try to avoid dashes in the App ID, since only the last part of the domain can contain one. If you would add extensions later, you would have using some like `org.qgis.qgis_ltr.Extension`.

I would use `org.qgis.qgis.LTR` as an ID.

**Comment by aleixpol** [comment]
I created an lts branch of it, I'd say we should go with that and close this PR

**Comment by eszlari** [comment]
> The proposed build for LTR version fails, because it's now searching for qgis-ltr executable(desktop icon) instead of qgis. I have no idea how to solve this.

You need to add `rename-desktop-file` (and `rename-appdata-file`) to the manifest to adjust the ID.

**Comment by eszlari** [comment]
> I created an lts branch of it, I'd say we should go with that and close this PR

Depends on if @hristo-mavrodiev wants to use both version in parallel easily.

Also: AFAIK this branch is not exposed in gnome-software, plasma-discover or flathub.org at the moment.

**Comment by aleixpol** [comment]
Discover should, and I don't see why other clients wouldn't.

**Comment by barthalion** [comment]
Aleix decided to use a branch so we're going to stick with it.

**Pull request closed without merging**